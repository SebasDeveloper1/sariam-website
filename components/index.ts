import Button from './Buttons/Button/Button';
import IconButton from './Buttons/IconButton/IconButton';
import SliderComponent from './SliderComponent/SliderComponent';
import ServiceCard from './ServiceCard/ServiceCard';
export { Button, IconButton, SliderComponent, ServiceCard };
