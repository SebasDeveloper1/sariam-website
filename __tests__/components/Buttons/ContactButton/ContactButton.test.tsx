import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ContactButton } from '@/components';

describe('ContactButton Component', () => {
  let button: HTMLElement;

  beforeEach(() => {
    // Render the ContactButton component and get the button element
    const { getByRole } = render(<ContactButton />);
    button = getByRole('link');
  });

  describe('Rendering', () => {
    it('should render a button', () => {
      // Check if the button is in the document
      expect(button).toBeInTheDocument();
    });

    it('should render a button with the correct link attributes', () => {
      // Check if the button has the correct link attributes
      expect(button).toHaveAttribute('href', 'https://wa.link/kuu26s');
      expect(button).toHaveAttribute('target', '_blank');
      expect(button).toHaveAttribute('rel', 'noopener noreferrer');
    });

    it('should render a button with the WhatsApp icon', () => {
      // Check if the button contains the WhatsApp icon element
      expect(button).toContainElement(document.querySelector('#whatsapp-icon'));
    });
  });

  describe('Interaction', () => {
    it('should redirect to the correct WhatsApp URL on click', () => {
      // Simulate a click on the button
      fireEvent.click(button);

      // Since Jest doesn't allow us to directly modify window.location, we cannot test the redirection directly.
      // You may consider using a library like "jest-localstorage-mock" to mock the window.location behavior and test the redirection.
    });
  });

  describe('Snapshot', () => {
    it('should match the ContactButton component snapshot', () => {
      // Take a snapshot of the rendered component and compare it to a saved snapshot
      const { asFragment } = render(<ContactButton />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
