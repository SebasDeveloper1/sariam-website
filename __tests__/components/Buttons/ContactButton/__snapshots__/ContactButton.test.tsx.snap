// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactButton Component Snapshot should match the ContactButton component snapshot 1`] = `
<DocumentFragment>
  <div
    class="button-primary w-1/2 h-14 bg-gray-800 hover:bg-gray-800 animate-pulse"
  />
</DocumentFragment>
`;
